src/proteingym/base/measurement.py

```diff
@@ -0,0 +1,194 @@
+    """The assay name to which the measurements belong."""
+
+    path: FilePath
+    """The path to the assay file, csv only."""
+
+    description: str | None = None
+    """A brief description"""
+
+    fields: list[Field]
+    """The list of fields in the measurement manifest."""
+
+    @field_validator("path", mode="before", check_fields=True)
+    def validate_path(cls, path: Path, info: ValidationInfo) -> Path:
+        """Optionally, extend the path with the `relative_to_path` from the context."""
+        if info.context and info.context.get("relative_to_path"):
+            path = info.context["relative_to_path"] / path
+        fmt = path.suffix[1:].lower()
+        if fmt not in MeasurementsFormat:
+            raise ValueError(f"Unsupported measurements file format: {fmt}")
+        return path
+
+    @field_serializer("path", check_fields=True)
+    def serialize_path(self, path: Path, info: SerializationInfo) -> str:
+        """Serialize the path as a Posix path."""
+        if info.context and info.context.get("relative_to_path"):
+            path = path.relative_to(info.context["relative_to_path"])
+        return path.as_posix()
+
+    @model_validator(mode="after")
+    def validate_field_names(self) -> "MeasurementsManifestSection":
+        """Validate whether field names are present in the `path` file."""
+        with self.path.open("r") as f:
+            header = f.readline()
+        for field in self.fields:
+            if field.name not in header:
+                raise ValueError(
+                    f"Field '{field.name}' not found in the file: {self.path}"
+                )
+        return self
+
+
+@dataclasses.dataclass(kw_only=True, frozen=True)
+class Measurements:
+    """The measurements on which an assay is based."""
+
+    name: str
+    """The name of the measurement."""
+
+    description: str | None = None
+    """A brief description"""
+
+    fields: list[Field] = dataclasses.field(default_factory=list)
+    """The measurement fields."""
+
+    records: list[tuple[str | int | float | bool | str, ...]] = dataclasses.field(
+        default_factory=list
+    )
+    """The measurement records."""
+
+    @classmethod
+    def from_manifest_section(
+        cls,
+        section: MeasurementsManifestSection,
+    ) -> "Measurements":
+        """Creates Measurements from a manifest section.
+
+        Args:
+            section (MeasurementsManifestSection): The manifest section
+                describing the measurements.
+
+        Returns:
+            Measurements: The created Measurements object.
+        """
+        columns = [field.name for field in section.fields]
+        # Reusing polars as we already depend on it for assays
+        records = list(pl.read_csv(section.path, columns=columns).iter_rows())
+        return cls(
+            name=section.name,
+            records=records,
+            fields=section.fields,
+            description=section.description,
+        )
+
```

Comment:
afford a `to_df` here as well - same thing as the one from assay but this time it's just a row-concatenation - and i guess a adding a column `name` to reflect the AssayRaw it came from